Remotion/Core/Core/Reflection/TypeDiscovery/BaseTypeCache.cs

```diff
@@ -69,13 +70,19 @@ public static BaseTypeCache Create (IEnumerable<Type> types)
             } while (nextBaseType != null);
           }
 
-          foreach (var interfaceType in type.GetInterfaces())
+          var implementedInterfaces = type.GetInterfaces();
+          if (implementedInterfaces.Length > 0)
```

Comment:
Optimization vs code question:
Is there a benefit in guarding the entry? The clear-operation and the loop aren't really an issue for empty lists but we got more conditions so the code is more complex. thoughts?

Reply:
I checked and removed the if